src/main/scala/org/ergoplatform/network/ErgoNodeViewSynchronizer.scala

```diff
@@ -1220,16 +1217,36 @@ object ErgoNodeViewSynchronizer {
       */
     case class FailedOnRecheckTransaction(id : ModifierId, error: Throwable) extends ModificationOutcome
 
-    case class RecoverableFailedModification(modifier: BlockSection, error: Throwable) extends ModificationOutcome
+    case class RecoverableFailedModification(typeId: ModifierTypeId, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
 
-    case class SyntacticallyFailedModification(modifier: BlockSection, error: Throwable) extends ModificationOutcome
+    case class SyntacticallyFailedModification(typeId: ModifierTypeId, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
 
-    case class SemanticallyFailedModification(modifier: BlockSection, error: Throwable) extends ModificationOutcome
+    case class SemanticallyFailedModification(typeId: ModifierTypeId, modifierId: ModifierId, error: Throwable) extends ModificationOutcome
 
-    case class SyntacticallySuccessfulModifier(modifier: BlockSection) extends ModificationOutcome
+    case class SyntacticallySuccessfulModifier(typeId: ModifierTypeId, modifierId: ModifierId) extends ModificationOutcome
 
-    case class SemanticallySuccessfulModifier(modifier: BlockSection) extends ModificationOutcome
+    /**
+      * Signal sent by node view holder when a full block is applied to state
+      * @param header - full block's header
+      */
+    case class FullBlockApplied(header: Header) extends ModificationOutcome
 
+    /**
+      * Signal sent after block sections processing (validation and application to state) done
+      * @param headersCacheSize - headers cache size after processing
+      * @param blockSectionsCacheSize - block sections cache size after processing
+      * @param cleared - blocks removed from cache being overfull
+      */
+    case class BlockSectionsProcessingCacheUpdate(headersCacheSize: Int,
+                                                  blockSectionsCacheSize: Int,
+                                                  cleared: (ModifierTypeId, Seq[ModifierId]))
+
+    /**
+      * Command to re-check mempool to clean transactions become invalid while sitting in the mempool up
```

Comment:
What is the difference between this event and the FullBlockApplied? Please reply in the code comment.

Reply:
I do not understand why to write this specifically in a comment . Events are used differently. Further can be combined likely, but this is out of scope of the PR.